src/plugins/vis_type_timeseries/server/lib/vis_data/request_processors/annotations/date_histogram.js

```diff
@@ -24,7 +24,7 @@ export function dateHistogram(
 ) {
   return (next) => async (doc) => {
     const barTargetUiSettings = await uiSettings.get(UI_SETTINGS.HISTOGRAM_BAR_TARGET);
-    const timeField = annotation.time_field;
+    const timeField = annotation.time_field || indexPattern?.timeFieldName || '';
```

Comment:
We merged #95772. This PR should be updated and retested. At least `indexPattern -> seriesIndex.indexPattern`

Reply:
Updated and retested